contentpacks/khanacademy.py

```diff
@@ -516,147 +575,147 @@ def download_and_clean_kalite_data(url, path, lang="en") -> str:
 
-    if not lang == en_lang_code and not no_dubbed_videos:
-        # Generate en_nodes.json json this will be used in dubbed video mappings.
-        # This will cache en_nodes.json
-        url = lang_url.format(projection=json.dumps(projection), lang=en_lang_code, ka_domain=ka_domain)
-        download_and_clean_kalite_data(url, lang=en_lang_code, ignorecache=False, filename="en_nodes.json")
+    Example 1: Swahili language { 
+                "sw":{ "name":"Swahili", "native_name":"Kiswahili" },
+                "swa":{ "name":"Swahili", "native_name":"Kiswahili" },
+            }
 
-        node_data = addin_dubbed_video_mappings(node_data, lang)
+    Example 2: Somali language {
+                "som":{ "name":"Somali", "native_name":"Soomaaliga" },
+                "so":{ "name":"Somali", "native_name":"Soomaaliga, af Soomaali" },
+            }
+    """
 
+    # MUST: Get all possible language codes for the language.
+    lang_codes = get_lang_code_list(lang)
+
+    # Loop-thru all lang codes and populate the topic, exercise, video lists while checking for duplicates.
+    logging.info("Found %s language codes for the language %s." % (lang_codes, lang,))
+    for lang_code in lang_codes:
+        logging.info("  Processing language code %s..." % lang_code)
+        projection = json.dumps(PROJECTION_KEYS)
+        url = API_URL.format(projection=projection, lang=lang_code, ka_domain=ka_domain)
+        node_data_path = download_and_clean_kalite_data(url, lang=lang_code, ignorecache=force, filename="nodes.json")
+        with open(node_data_path, 'r') as f:
+            node_data_temp = ujson.load(f)
+        """
+        Some translated_youtube_lang value return from KHAN API did not match
+            to the specify language code. We need to override it to use the same
+            language code.
+        Example: using pt-BR language code in the khan api will return pt translated_youtube_lang.
+        """
+        for node_temp in node_data_temp:
+            node_kind = node_temp.get("kind")
+            if (node_kind == NodeType.topic):
+                if not node_temp["path"] in topic_path_list:
+                    topic_path_list.append(node_temp["path"])
+                    node_data.append(node_temp)
+            if (node_kind == NodeType.exercise):
+                if not node_temp["id"] in exercise_ids:
+                    exercise_ids.append(node_temp["id"])
+                    node_data.append(node_temp)    
+            if (node_kind == NodeType.video):
+                if not node_temp["youtube_id"] in youtube_ids:
+                    youtube_lang = node_temp["translated_youtube_lang"]
+                    if youtube_lang == lang:
+                        youtube_ids.append(node_temp["youtube_id"])
+                        node_data.append(node_temp)
+                    elif not youtube_lang != EN_LANG_CODE:
```

Comment:
I assume that it also means @mrpau-richard isn't using a PEP8 checker in his editor? :)